lib/flip.test.js

```diff
@@ -3,11 +3,14 @@ const test = require('ava')
 const flip = load('./flip.mjs').default
 
 const add = (a, b) => a + b
+const add4 = (a, b, c, d) => a + b + c + d
 
 test('flip', t => {
-  t.is(flip(add)('x')('y'), 'yx', 'should work passing argument in individually')
-  t.is(flip(add, 'a', 'b'), 'ba', 'should work passing all arguments at once')
-  t.is(flip(add, 'c')('d'), 'dc', 'should work passing some arguments straight away')
-  t.is(flip(add)('e', 'f'), 'fe', 'should work passing in arguments later')
-  t.is(flip(add, 'x', 'y', 'z'), 'yx', 'should ignore additional arguments')
+  t.is(flip(add)('x')('y'), 'yx', 'flips after accepting individual arguments')
+  t.is(flip(add, 'a', 'b'), 'ba', 'flips after accepting all arguments at once')
+  t.is(flip(add, 'c')('d'), 'dc', 'flips after accepting some arguments straight away')
+  t.is(flip(add)('e', 'f'), 'fe', 'flips after only function first, with batched arugments later')
+  t.is(flip(add, 'x', 'y', 'z'), 'yx', 'ignores extraneous arguments')
```

Comment:
I would expect the following to be treated the same, but these tests suggests that this isn't happening. Either we should ignore additional arguments in all cases, or include them in all cases.
```javascript
flip(add, 'a', 'b', 'c')
flip(add)('a', 'b', 'c')
flip(add)('a')('b')('c')
```
```

Reply:
I don't understand. Please try to clarify what you mean. I only added 1 test at the end, the rest you wrote. I just changed the message text.

Reply:
Oh nevermind. I see what you're getting at. I should not be adding an endpoint to the last slice in my patch.

Reply:
Hmm. I've made this edit to pass on the extra arguments and can push the changes if you want. But to match the behavior in flipAll is kind of gross looking.

I'm fine with just dropping the extraneous arguments, which is what the current code of this PR does in both `flip` and `flipAll`. To actually make use of any extra arguments, user would need to write functions that use `[].slice(arguments)` to get at them. Any named parameters would change the `length` of the function and thus not be extraneous.

Reply:
I dropping the extra arguments would keep the code simpler and consistent.